tests/tpch/generate-data.py

```diff
@@ -1,47 +1,145 @@
+import functools
 import pathlib
 import tempfile
+import warnings
 
 import botocore.session
 import click
+import coiled
 import dask
 import duckdb
 import psutil
+import pyarrow.compute as pc
+from dask.distributed import LocalCluster, wait
 
 
 def generate(
-    scale: int = 10, partition_size: str = "50 MiB", path: str = "./tpch-data"
+    scale: int = 10,
+    partition_size: str = "128 MiB",
+    path: str = "./tpch-data",
+    relaxed_schema: bool = False,
 ):
     if str(path).startswith("s3"):
         path += "/" if not path.endswith("/") else ""
-        path += f"scale-{scale}/"
+        path += f"scale-{scale}{'-strict' if not relaxed_schema else ''}/"
+        use_coiled = True
     else:
-        path = pathlib.Path(path) / f"scale-{scale}/"
+        path = (
+            pathlib.Path(path)
+            / f"scale-{scale}{'-strict' if not relaxed_schema else ''}/"
+        )
         path.mkdir(parents=True, exist_ok=True)
+        use_coiled = False
 
     print(f"Scale: {scale}, Path: {path}, Partition Size: {partition_size}")
+    kwargs = dict(
+        scale=scale,
+        path=path,
+        relaxed_schema=relaxed_schema,
+        partition_size=partition_size,
+    )
+
+    if use_coiled:
+        with coiled.Cluster(
+            n_workers=10,
+            worker_memory="4 GiB",
```

Comment:
We don't have to tune this too much but this gives us T3 instances. It may be actually cheaper to just use an 8GiB M* instance. T3 sucks hard for the stuff we're doing (I'm just thinking about >10k scales here. Everything else is already negligible in cost)

(https://github.com/coiled/platform/issues/3652)

Reply:
There is also the option of using an ARM machine with `m7g.medium` which gives us one CPU and 4GiB of RAM

Reply:
ARM was more temperamental complaining about specifying all deps at the same time, and then turned out to only have duckdb 0.7.1 available (not 0.9.1). 

I've set the vm types to m7a.medium, and m3.medium each of which has 1vCPU and ~4GiB and [works very nicely for the current tasks](https://cloud.coiled.io/clusters/299433/information?account=dask-engineering&tab=Metrics) and also ended up being ~2 minutes faster than a previous scale-100 run with T3 instances: https://cloud.coiled.io/clusters/299216/information?account=dask-engineering